### ISSUE TITLE
Investigate failing tests on Linux and MacOS when upgrading MeshKernel to 2.0.0

### ISSUE THREAD

**Issue by priscavdsluis** [Deltares/HYDROLIB-core#444]
**What is the need for this task.**
We are trying to upgrade MeshKernel from 1.0.0 to 2.0.0, see PR #442.
When doing that we notice that the CI is failing on Linux and MacOS:
![CI-jobs](https://user-images.githubusercontent.com/36264671/214841033-6d51101e-f612-47ef-8422-a0b4f71f0a57.png)

**What is the task?**
Investigate why these tests are failing and whether it is on the hydrolib-core or meshkernel side.

**Additional context**
The stacktrace: [stacktraces.txt](https://github.com/Deltares/HYDROLIB-core/files/10509442/stacktraces.txt)

**Comment by backeb** [comment]
**Describe bug:**
failed to install `dfm_tools` on Mac OSX

**OS**
`Darwin Bjorns-MacBook-Pro-2.local 21.6.0 Darwin Kernel Version 21.6.0: Mon Dec 19 20:43:09 PST 2022; root:xnu-8020.240.18~2/RELEASE_ARM64_T6000 arm64`

**conda --version**
`conda 23.1.0`

**pip --version**
`pip 23.0.1 from /Users/backeb/mambaforge/envs/dfm_tools_env/lib/python3.8/site-packages/pip (python 3.8)`

**Minimal example**
```
conda create --name dfm_tools_env -c conda-forge python=3.8
conda activate dfm_tools_env
conda install -c conda-forge git shapely cartopy pyepsg geopandas contextily xarray dask netcdf4 bottleneck xugrid cdsapi pydap
python -m pip install git+https://github.com/Deltares/dfm_tools
```

**Error message** on exactuting `python -m pip install git+https://github.com/Deltares/dfm_tools`
```
Collecting pydantic<1.11,>=1.10
  Using cached pydantic-1.10.6-cp38-cp38-macosx_11_0_arm64.whl (2.5 MB)
Collecting meshkernel<3.0.0,>=2.0.0
  Using cached meshkernel-2.0.2.tar.gz (364 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [18 lines of output]
      Traceback (most recent call last):
        File "/Users/backeb/mambaforge/envs/dfm_tools_env/lib/python3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/backeb/mambaforge/envs/dfm_tools_env/lib/python3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/Users/backeb/mambaforge/envs/dfm_tools_env/lib/python3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/private/var/folders/0d/d0d2xvbs4p1061xfbpnzrt040000gn/T/pip-build-env-40rtvd7u/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 338, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/private/var/folders/0d/d0d2xvbs4p1061xfbpnzrt040000gn/T/pip-build-env-40rtvd7u/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 320, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/0d/d0d2xvbs4p1061xfbpnzrt040000gn/T/pip-build-env-40rtvd7u/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 484, in run_setup
          super(_BuildMetaLegacyBackend,
        File "/private/var/folders/0d/d0d2xvbs4p1061xfbpnzrt040000gn/T/pip-build-env-40rtvd7u/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 335, in run_setup
          exec(code, locals())
        File "<string>", line 222, in <module>
        File "<string>", line 70, in get_library_name
      OSError: Unsupported operating system: Darwin
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Additional context**
Using terminal in `Visual Studio Code`
```
Version: 1.76.2 (Universal)
Commit: ee2b180d582a7f601fa6ecfdad8d9fd269ab1884
Date: 2023-03-14T17:54:09.061Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Darwin arm64 21.6.0
Sandboxed: No
```

**Comment by lucacarniato** [comment]
Meshkernel is available as a compiled library in the wheel packages for both Windows and Linux. To install it, simply run the command 'pip install meshkernel'. There's no need to build Meshkernel from sources for these operating systems.

However, for Mac, there is currently no solution available as we don't have macOS agents. The Meshkernel 1.0.0 wheel package did include a compiled macOS library that was generated by one of our developers on their laptop.

**Comment by backeb** [comment]
Thanks @lucacarniato 

So you're saying I can install `dfm_tools` by doing:
1. `conda create --name dfm_tools_env -c conda-forge python=3.8`
2. `conda activate dfm_tools_env`
3. `conda install -c conda-forge git shapely cartopy pyepsg geopandas contextily xarray dask netcdf4 bottleneck xugrid cdsapi pydap`
4. `pip install meshkernel`
5. `python -m pip install git+https://github.com/Deltares/dfm_tools`

**Comment by lucacarniato** [comment]
installing meshkernel before dfm_tools could work

**Comment by backeb** [comment]
Hi @lucacarniato,

When I follow the steps in https://github.com/Deltares/HYDROLIB-core/issues/444#issuecomment-1488542448 on a Mac, I get this error:
```
(dfm_tools_env) backeb@Bjorns-MacBook-Pro-2:bed-data-preparation>pip install meshkernel                                                                                                                                    
                                                                                                                                                                                                                           
Collecting meshkernel                                                                                                                                                                                                      
  Using cached meshkernel-2.0.2.tar.gz (364 kB)                                                                                                                                                                            
  Installing build dependencies ... done                                                                                                                                                                                   
  Getting requirements to build wheel ... error                                                                                                                                                                            
  error: subprocess-exited-with-error                                                                                                                                                                                      
                                                                                                                                                                                                                           
  × Getting requirements to build wheel did not run successfully.                                                                                                                                                          
  │ exit code: 1                                                                                                                                                                                                           
  ╰─> [18 lines of output]                                                                                                                                                                                                 
      Traceback (most recent call last):                                                                                                                                                                                   
        File "/Users/backeb/mambaforge/envs/dfm_tools_env/lib/python3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>                                                       
          main()                                                                                                                                                                                                           
        File "/Users/backeb/mambaforge/envs/dfm_tools_env/lib/python3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main                                                           
          json_out['return_val'] = hook(**hook_input['kwargs'])                                                                                                                                                            
        File "/Users/backeb/mambaforge/envs/dfm_tools_env/lib/python3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel                                   
          return hook(config_settings)                                                                                                                                                                                     
        File "/private/var/folders/0d/d0d2xvbs4p1061xfbpnzrt040000gn/T/pip-build-env-sjs_ibft/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel                     
          return self._get_build_requires(config_settings, requirements=['wheel'])                                                                                                                                         
        File "/private/var/folders/0d/d0d2xvbs4p1061xfbpnzrt040000gn/T/pip-build-env-sjs_ibft/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires                              
          self.run_setup()                                                                                                                                                                                                 
        File "/private/var/folders/0d/d0d2xvbs4p1061xfbpnzrt040000gn/T/pip-build-env-sjs_ibft/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 487, in run_setup                                        
          super(_BuildMetaLegacyBackend,                                                                                                                                                                                   
        File "/private/var/folders/0d/d0d2xvbs4p1061xfbpnzrt040000gn/T/pip-build-env-sjs_ibft/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 338, in run_setup                                        
          exec(code, locals())                                                                                                                                                                                             
        File "<string>", line 222, in <module>                                                                                                                                                                             
        File "<string>", line 70, in get_library_name                                                                                                                                                                      
      OSError: Unsupported operating system: Darwin                                                                                                                                                                        
      [end of output]                                                                                                                                                                                                      
                                                                                                                                                                                                                           
  note: This error originates from a subprocess, and is likely not a problem with pip.                                                                                                                                     
error: subprocess-exited-with-error                                                                                                                                                                                        
                                                                                                                                                                                                                           
× Getting requirements to build wheel did not run successfully.                                                                                                                                                            
│ exit code: 1                                                                                                                                                                                                             
╰─> See above for output.                                                                                                                                                                                                  
                                                                                                                                                                                                                           
note: This error originates from a subprocess, and is likely not a problem with pip.      
```

Will test on Linux now.

**Comment by backeb** [comment]
Following [this](https://github.com/Deltares/HYDROLIB-core/issues/444#issuecomment-1488542448) approach on a CentOS 8 VM, I managed to install `dfm_tools`.

**Comment by arthurvd** [comment]
Hi @backeb, regarding your original error: "OSError: Unsupported operating system: Darwin", indeed, the latest MeshKernel(Py) does not come in a MacOSX version. I've already listed it in the hydrolib-core release notes from previous month:
https://deltares.github.io/HYDROLIB-core/0.5.2/releasenotes/#compatibility-notes
But maybe @lucacarniato can list it somewhere on the meshkernel pages as well.

**Comment by backeb** [comment]
Thanks @arthurvd, I'm working on a docker container, so I can use dfm_tools from my Mac.

**Comment by backeb** [comment]
@arthurvd I've tried some steps to getting it working in an ubuntu docker container here: https://github.com/openearth/bed-data-preparation/issues/1#issuecomment-1545822906 but the install fails on a `cmake` dependency.

**Comment by veenstrajelmer** [comment]
Meshkernelpy 2.1.0 was released a few weeks ago, macos/linux support was fixed here. This issue can probably be fixed by updating the dependencies.

**Comment by veenstrajelmer** [comment]
Will be fixed with:
- https://github.com/Deltares/MeshKernelPy/pull/112
- https://github.com/Deltares/MeshKernelPy/pull/130
- https://github.com/Deltares/MeshKernelPy/pull/131

In the next release of meshkernel

**Comment by priscavdsluis** [comment]
If macOS support is fixed, we should re-add the macOS in our GitHub ci.yml workflow.

**Comment by veenstrajelmer** [comment]
Update after adding macos to the github workflow:
![image](https://github.com/Deltares/HYDROLIB-core/assets/60435591/36bcc917-a7f6-41c5-bef9-41ef479e017d)

Installation succeeds, but there is one failing testcase:
``` 
FAILED tests/dflowfm/test_net.py::test_create_1d_2d_1d2d
```

**Comment by veenstrajelmer** [comment]
@backeb This is a mamba issue, it tries to solve (and it fails of course) and in that solving process it tries to install older and older versions of dfm_tools, hydrolib-core and meshkernel. This old meshkernel version does not have macOS support, so it makes sense that it fails. Please try to run the code in https://github.com/Deltares/HYDROLIB-core/issues/444#issuecomment-2078905732 so we can swiftly fix macos support in HYDROLIB-core.

**Comment by backeb** [comment]
> @backeb This is a mamba issue, it tries to solve and in that solving process it tries to install older and older versions of dfm_tools, hydrolib-core and meshkernel. Mamba is clearly not smart enough to solve this. This old meshkernel version does not have macOS support, so it makes sense that it fails. Please try to run the code in [#444 (comment)](https://github.com/Deltares/HYDROLIB-core/issues/444#issuecomment-2078905732) so we can swiftly fix macos support in HYDROLIB-core.

Seems to be an issue with installing meshkernelpy: https://github.com/Deltares/MeshKernelPy/issues/162

**Comment by veenstrajelmer** [comment]
After some trial and error I noticed that the `macos-12` and `macos-13` testbanks do run (both seem to be x86_64), so the failing testcase seems to be `macos-14` (arm64). Also, the installation issues above seem to be meshkernel related, since it does not cover macos-13-arm64 since we do not have these runners available. Therefore, I suggest to wrap up this issue (which is to repair linux and macos testbenches) by testing on macos-13 now. I have created a follow-up issue to investigate macos 14 arm64 support here: https://github.com/Deltares/HYDROLIB-core/issues/635
